sqs/sqs_test.go

```diff
@@ -0,0 +1,47 @@
+package sqs
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/stretchr/testify/assert"
+)
+
+type MockSQS struct {
+	QueueAttributes *sqs.GetQueueAttributesOutput
+}
+
+func (m *MockSQS) GetQueueAttributes(*sqs.GetQueueAttributesInput) (*sqs.GetQueueAttributesOutput, error) {
+	return m.QueueAttributes, nil
+}
+
+func (m *MockSQS) SetQueueAttributes(input *sqs.SetQueueAttributesInput) (*sqs.SetQueueAttributesOutput, error) {
+	m.QueueAttributes = &sqs.GetQueueAttributesOutput{
+		Attributes: input.Attributes,
+	}
+
+	return &sqs.SetQueueAttributesOutput{}, nil
+}
+
+func NewMockSqsClient() *SqsClient {
+	Attributes := make(map[string]*string)
+	Attributes["ApproximateNumberOfMessages"] = aws.String("50")
+
+	return &SqsClient{
+		Client: &MockSQS{
+			QueueAttributes: &sqs.GetQueueAttributesOutput{
+				Attributes: Attributes,
+			},
+		},
+		QueueUrl: "sqs.test.com",
```

Comment:
when using example urls, you should really use example.com which is reserved by iana for exactly this purpose